.buildkite/scripts/publish.sh

```diff
@@ -22,9 +22,9 @@ else
     DOCKER_NAMESPACE="${DOCKER_IMG_PR}"
 fi
 
-TAG_NAME="${GITHUB_PR_BRANCH}"
+TAG_NAME="${BUILDKITE_BRANCH//\:/-}" # forked branches have names: "<account>:<name-of-the-branch>"
 # if tag exists use tag instead
-if [[ -n ${BUILDKITE_TAG} ]]; then
+if [[ -n ${BUILDKITE_TAG:-} ]]; then
     TAG_NAME=${BUILDKITE_TAG}
 fi
```

Comment:
Currently, in Jenkins when it is a Pull Request the docker tags that are pushed are:
- the commit from the pull request (e.g. ``)
  `docker push docker.elastic.co/observability-ci/package-registry:8f2ea87b40909477d4059f6ef8162e0aeef79c52` 
- the PR number:
  `docker push docker.elastic.co/observability-ci/package-registry:PR-1034`

Maybe it can be added `TAG_NAME` to be set in the first `if` about whether or not this is a pull request. 

Not tested, I would suggest this change: 
```shell
if [[ "${BUILDKITE_PULL_REQUEST}" == "false" ]]; then
    DOCKER_NAMESPACE="${DOCKER_IMG}"
    TAG_NAME="${BUILDKITE_BRANCH}"  # e.g. main
else
    DOCKER_NAMESPACE="${DOCKER_IMG_PR}"
    TAG_NAME="PR-${BUILDKITE_PULL_REQUEST}"
fi

# if tag exists use tag instead
if [ -n "${BUILDKITE_TAG}" ]; then
    TAG_NAME="${BUILDKITE_TAG}"
fi

DOCKER_IMG_TAG="${DOCKER_NAMESPACE}:${BUILDKITE_COMMIT}"
DOCKER_IMG_TAG_BRANCH="${DOCKER_NAMESPACE}:${TAG_NAME}"

pushDockerImage
```